axon-framework/queries/query-dispatchers.md

```diff
@@ -227,5 +227,132 @@ commandGateway.sendAndWait(new RedeemCmd("gc1", amount));
+
+
+Streaming query is flexible enough to be used with **any** query return type. That means that any return type that is not Flux will automatically be converted to Flux, based on that Flux will return single or multiple items.
+
+Query gateway provides nifty `streamingQuery` method to utilize streaming query easily. 
+It's simple to use, and requires just two parameters: query and expected response type class.
+`streamingQuery` method **is lazy**, meaning query is sent once Flux is subscribed to.
+
+
+It is also possible to utilize streaming query with `query` method: `queryGateway.query(query,  ResponseTypes.fluxOf(CardSummary.class))`.
+However, this approach is **eager** and less intuitive as it requires more boilerplate code.
+In this case `query` method will send query immediately and return CompletableFuture which returns a payload of type Flux, that needs to be subscribed to.
+
+```java
+@QueryHandler
+public List<CardSummary> handle(FetchCardSummariesQuery query) {
+        ...
+    return cardRepository.findAll(); //1
+}
+        ...
+
+public Flux<CardSummary> consumer() {
+        return queryGateway.streamingQuery(query, CardSummary.class); //2
+}
+```
+
+1. We are querying the `cardRepository` for all the cards, that can potently return a large result set containing thousands of items.
+2. We are using the `queryGateway` to issue the query. If we used `multipleInstanceOf(CardSummary.class)` we would get large list transferred as single result message over the network, potentially causing a buffer overflow and max message size violation.
+Instead, we are using `streamingQuery(query, CardSummary.class)`, that will convert our response to Flux stream and chunk the result set into smaller messages of CardSummary.
+
+Natively, if we want fine-grained control of the producing stream we can use Flux as return type:
+
+```java
+@QueryHandler
+public Flux<CardSummary> handle(FetchCardSummariesQuery query) {
+        ...
+    return reactiveCardRepository.findAll(); 
+}
+```
+
+When using Flux as return type, we can control backpressure, stream cancellation and implement more complex features like pagination.
+
+> **Note**
+>
+> Backpressure and stream cancellation features are only available with Axon Server 4.6 + version. Streaming queries used with pre Axon Server 4.6 version will work, but without these important features.
+>
+
+> **Transaction leaking**
+>
+> Once consumer of streaming query receives Flux to subscribe to, transaction will be considered completed successfully. That means, that any subsequent messages on the stream will not be part of transaction. 
+> This includes errors, and as transaction is already over, error will not be propagated to parent transaction to invoke any rollback method.
+> This has implication that streaming query should not be used within Unit Of Work (within message handlers or any other transactional methods) to chaining other transactional actions (like sending a command or query).
+>
```

Comment:
Next to the suggestion, I'd also link to the Unit of Work page in the reference guide for added clarity.

```suggestion
> **Transaction Leaking Concerns**
>
> Once a consumer of the streaming query receives the `Flux` to subscribe to, the transaction will be considered completed successfully. 
> That means that any subsequent messages on the stream will not be part of the transaction, including errors.
> As the transaction is already over an error will not be propagated to the parent transaction to invoke any rollback method.
> This has the implication that the streaming query should not be used within a Unit Of Work (within message handlers or any other transactional methods) to chain other transactional actions (like sending a command or query).
```